src/dsp_tools/utils/request_utils.py

```diff
@@ -192,3 +196,27 @@ def should_retry(response: Response) -> bool:
     try_again_later = "try again later" in response.text.lower()
     in_testing_env = os.getenv("DSP_TOOLS_TESTING") == "true"  # set in .github/workflows/tests-on-push.yml
     return (try_again_later or in_500_range) and not in_testing_env
+
+
+def log_and_raise_request_exception(error: RequestException) -> Never:
+    msg = (
+        f"During an API call the following exception occurred, "
+        f"please contact the dsp-tools development team (at info@dasch.swiss) if you need help resolving the error.\n"
+        f"Original request: {error.request}\n"
```

Comment:
Printing `error.request`  will be ugly and not helpful. Better: 

```suggestion
        f"Original request: {error.request.method} {error.request.url}\n"
```